src/app/login/login.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+  rForm: FormGroup;
+  post: any;                     // A property for our submitted form
+  password: string = '';
+  name: string = '';
+  titleAlert: string = 'This field is required';
+  constructor(private fb: FormBuilder) {
+      this.rForm = fb.group({
+      'name' : [null, Validators.required],
+//      'password' : [null, Validators.compose([Validators.required, Validators.minLength(30), Validators.maxLength(500)])],
+      'password' : [null, Validators.required],
+      'validate' : ''
+    });
+  }
+
+  ngOnInit() {
+    this.rForm.get('validate').valueChanges.subscribe(
+
+      (validate) => {
+
+          if (validate === '1') {
```

Comment:
remove all this. As discussed, there are not needed

Reply:
removed